src/io/input_split_base.cc

```diff
@@ -228,20 +226,21 @@ bool InputSplitBase::ReadChunk(void *buf, size_t *size) {
                             max_size - olen);
   nread += olen;
   if (nread == 0) return false;
-  if (nread != max_size) {
-    *size = nread;
-    return true;
-  } else {
-    const char *bptr = reinterpret_cast<const char*>(buf);
-    // return the last position where a record starts
-    const char *bend = this->FindLastRecordBegin(bptr, bptr + max_size);
-    *size = bend - bptr;
-    overflow_.resize(max_size - *size);
-    if (overflow_.length() != 0) {
-      std::memcpy(BeginPtr(overflow_), bend, overflow_.length());
-    }
-    return true;
+  if (nread == olen) {  // add extra newline to handle files with NOEOL
+    char *bufptr = reinterpret_cast<char*>(buf);
+    bufptr[nread] = '\n';
```

Comment:
I don't think we'd want to add `\n` here, since line 194 is already adding one

Reply:
I removed lines 229-233 and ran two unit tests in `test/unittest/unittest_inputsplit.cc`. Both passed.

Reply:
I don't think so. It would get failed when the last the file is NOEOL(that's why it passed your test case) if we remove lines 229-233.
It can be fixed in two ways.
(1) add '\n' just like lines 229-233 shows.
(2) or modify the function FindLastRecordBegin to handle this case.(Maybe it should not return the beginPtr when the data contains one character "\n" only). But I found that it needs to change several files so I choose the Solution One.

Reply:
You should try it on your end too: remove lines 229-233 and run InputSplit unit tests.

Reply:
@hcho3  You could test it by creating the test case with a single file with NOEOL and it will drop into an infinite loop if we remove lines 229-233 here.

Reply:
> creating the test case with a single file with NOEOL and it will drop into an infinite loop if we remove lines 229-233 here.

Is this with two machines?

Reply:
@htgeis Ah got it. I do get infinite loop with a single NOEOL file. I will add another unit test to account for this.

Reply:
@htgeis I added a unit test to test cases where the last file has NOEOL. Let me know what you think.